torch/distributions/binomial.py

```diff
@@ -112,6 +112,14 @@ def log_prob(self, value):
                           - log_factorial_n)
         return value * self.logits - log_factorial_k - log_factorial_nmk - normalize_term
 
+    def entropy(self):
+        total_count = int(self.total_count.max())
+        if not self.total_count.min() == total_count:
+            raise NotImplementedError("Inhomogeneous total count not supported by `entropy`.")
+
+        log_prob = self.log_prob(self.enumerate_support(False))
+        return -(torch.exp(log_prob) * log_prob).sum(0)
```

Comment:
1. this should be `.sum(-1)` rather than `.sum(0)`.
2.  I think it would be most consistent to return an expanded version:
```py
        log_prob = self.log_prob(self.enumerate_support(expand=False))
        entropy = -(torch.exp(log_prob) * log_prob).sum(-1)
        return entropy.expand(self.batch_shape)
```

Reply:
Hmm. I thought that `Binomial.enumerate_support()` function enumerates probabilities in the first dimension (= dimension 0).

https://github.com/pytorch/pytorch/blob/ae501a9727c16b8bc83c2fd7b40d9d801133ac91/torch/distributions/binomial.py#L115-L120

Therefore, I tried to take a summation over dimension 0.

Reply:
Thanks for explaining, that now makes sense!

Reply:
Could you add the `.expand()` line? I'm surprised your original `expand=False` logic isn't failing tests 🤔

Reply:
Oh I see, the expand is not needed because it happens in `self.log_prob(-)`. LGTM